tests/time_tests/test_runner/conftest.py

```diff
@@ -161,36 +175,123 @@ def validate_test_case(request, test_info):
     """Fixture for validating test case on correctness.
 
     Fixture checks current test case contains all fields required for
-    a correct work. To submit results to a database test case have
-    contain several additional properties.
+    a correct work.
     """
-    schema = {
+    schema = """
+    {
         "type": "object",
         "properties": {
             "device": {
                 "type": "object",
                 "properties": {
                     "name": {"type": "string"}
-                }},
+                },
+                "required": ["name"]
+            },
             "model": {
                 "type": "object",
                 "properties": {
                     "path": {"type": "string"}
-                }},
+                },
+                "required": ["path"]
+            }
         },
+        "required": ["device", "model"],
+        "additionalProperties": false
     }
-    if request.config.getoption("db_submit"):
-        # For submission data to a database some additional fields are required
-        schema["properties"]["model"]["properties"].update({
-            "name": {"type": "string"},
-            "precision": {"type": "string"},
-            "framework": {"type": "string"}
-        })
-    test_info["submit_to_db"] = True
+    """
+    schema = json.loads(schema)
+
     try:
         validate(instance=request.node.funcargs["instance"], schema=schema)
     except ValidationError:
-        test_info["submit_to_db"] = False
+        request.config.option.db_submit = False
+        raise
+    yield
+
+
+@pytest.fixture(scope="function")
+def prepare_db_info(request, test_info, executable, niter):
+    """Fixture for preparing and validating data to submit to a database.
+
+    Fixture prepares data and metadata to submit to a database. One of the steps
+    is parsing of build information from build manifest. After preparation,
+    it checks if data contains required properties.
+    """
+    FIELDS_FOR_ID = ['run_id', 'timetest', 'model', 'device', 'niter']
+
+    run_id = request.config.getoption("db_submit")
+    if not run_id:
+        yield
+        return
+
+    # add db_metadata
+    test_info["db_info"].update(json.loads(request.config.getoption("db_metadata")))
+
+    # add test info
+    info = {
+        "run_id": run_id,
+        "timetest": str(executable.stem),
+        "model": request.node.funcargs["instance"]["model"],
+        "device": request.node.funcargs["instance"]["device"],
+        "niter": niter,
+        "test_name": request.node.name,
+        "results": test_info["results"]
+    }
+    info['_id'] = hashlib.sha256(
+        ''.join([str(info[key]) for key in FIELDS_FOR_ID]).encode()).hexdigest()
+    test_info["db_info"].update(info)
+
+    # add manifest metadata
+    manifest_metadata = metadata_from_manifest(request.config.getoption("manifest"))
```

Comment:
Is there a way to prepare and validate metadata once on cmd options validation vs. on every submission?

Reply:
Good idea! Yes, I did it by adding `manifest_metadata` fixture with it's own schema